src/Features/Core/Portable/Ambiguity/AbstractAmbiguousTypeCodeFixProvider.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.AddImports;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using static Microsoft.CodeAnalysis.CodeActions.CodeAction;
+
+namespace Microsoft.CodeAnalysis.AmbiguityCodeFixProvider
+{
+    internal abstract class AbstractAmbiguousTypeCodeFixProvider : CodeFixProvider
+    {
+        protected abstract SyntaxNode GetAliasDirective(string typeName, ISymbol symbol);
+
+        public override FixAllProvider GetFixAllProvider() => null;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            // Innermost: We are looking for an IdentifierName. IdentifierName is sometimes at the same span as its parent (e.g. SimpleBaseTypeSyntax).
+            var diagnosticNode = root.FindNode(context.Span, getInnermostNodeForTie: true);
```

Comment:
"getInnermostNodeForTie: true" is correct.  But can you verify that this is working by writing a test case where your amibiguous type name is an argument to some method?  That way we can demonstrate that FindNode is finding the name and not he ArgumentSyntax node.

Reply:
I have some tests where getInnermostNodeForTie is relevant including [this one](https://github.com/dotnet/roslyn/pull/24022/files#diff-871cc5d400cdd54881c965ca53a634eeR448) for method arguments.